### PR TITLE
Add custom granularity spec to time spine

### LINKED ISSUE #335: [Feature] Custom calendar support
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward change to existing dbt-semantic-interfaces functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Support custom calendar spec in time spine.
```
time_spine:
  standard_granularity_column: date_day
  custom_granularity_columns:
      - name: fortnight
      - name: retail_month
      - name: fiscal_quarter
```

### Describe alternatives you've considered

_No response_

### Who will this benefit?

_No response_

### Are you interested in contributing this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by courtneyholcomb**
Resolves #335

### Description
Support custom granularity spec.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: tests/example_project_configuration.py [hunk 1]
```diff
@@ -26,6 +27,10 @@
         PydanticTimeSpine(
             node_relation=PydanticNodeRelation(alias="day_time_spine", schema_name="stuff"),
             primary_column=PydanticTimeSpinePrimaryColumn(name="ds_day", time_granularity=TimeGranularity.DAY),
+            custom_granularity_columns=[
+                PydanticTimeSpineCustomGranularityColumn(name="retail_year"),
+                PydanticTimeSpineCustomGranularityColumn(name="martian_week"),
```
Comment (tlento) [inline tests/example_project_configuration.py:32]: ted lasso yes

Comment (tlento) [reply tests/example_project_configuration.py:32]: I wonder how long a martian week is, though. 🤔 

Wikipedia knows, of course..... https://en.wikipedia.org/wiki/Darian_calendar

**Review by tlento — APPROVED** [review]
Seems reasonable. I filed a task to add validation, I think the names will have to follow our usual rules (no leading or trailing underscores, etc.) and they have to be unique, but we don't need it just yet.

File: tests/example_project_configuration.py (hunk 1 above)
Comment (courtneyholcomb) [reply tests/example_project_configuration.py:32]: LMAO

**Pull request merged**